src/components/TabsContent.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import { ListingItem } from "./ListingItem";
+import { useSelector } from "react-redux";
+import { ApplicationState } from "../app/store";
+import { ListItem } from "../types/ListItem";
+import { Tab } from "../types/Tab";
+
+export type TabsContentProps = {
+  displayTabs: Tab[];
+  selectedTabIndex: number;
+};
+export const TabsContent: (props: TabsContentProps) => JSX.Element = ({
+  displayTabs,
+  selectedTabIndex,
+}: TabsContentProps) => {
+  const { items } = useSelector((state: ApplicationState) => state);
+  return (
+    <div>
+      {displayTabs.map((tab) => {
+        let itemsToDisplay: ListItem[] = [];
+        if (tab.title === "to read") {
+          itemsToDisplay = items.filter((item) => !item.alreadyRead);
+        } else if (tab.title === "already read") {
+          itemsToDisplay = items.filter((item) => item.alreadyRead);
+        } else {
+          itemsToDisplay = items.filter((item) =>
+            item.tags.find((tag) => tag.title === tab.title)
+          );
+        }
+        return (
+          <div
+            role="tabpanel"
+            hidden={selectedTabIndex !== tab.index}
+            id={`vertical-tabpanel-${tab.index}`}
+            aria-labelledby={`vertical-tab-${tab.index}`}
+            key={tab.index}
+          >
+            <table>
```

Comment:
How about having something like:
`
{ itemsToDisplay.length && <table>...everything here without any conditional logic inside </table> }`

`{ !itemsToDisplay.length && <spanorsomething>Yo yo no items yet so add some :))))) </...> }
`

Reply:
If itemsToDisplay is empty, 

> 0

 is printed out. So I've used the ternary operator instead in order to print different layout when `itemsToDisplay` is empty.